unit_tests/vector_test/vector_test.cpp

```diff
@@ -675,6 +694,122 @@ TEST_F(VectorTests, ConstantTest) {
   }
 }
 
+TEST_F(VectorTests, EqualOperatorTest) {
```

Comment:
Rather than place this all in a single test, why not reuse the TEST_SCALAR_AND_INT_F() macro to reduce the amount of boilerplate code?  For example...

template<class T, int d>
void Equal_Test(const T& precision) {
  mathfu::Vector<T, d> expected;
  for (int i = 0; i < d; ++i) {
    expected[i] = static_cast<T>(i);
  }
  mathfu::Vector<T, d> copy = expected;
  EXPECT_TRUE(expected, copy);
}

TEST_SCALAR_AND_INT_F(Equal);